tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -446,26 +446,29 @@ private static void StartDiagnosticManager()
         private static void InitializeDebugger(TracerSettings tracerSettings)
         {
             var manager = DebuggerManager.Instance;
-            if (manager.DebuggerSettings.CodeOriginForSpansEnabled
-                || manager.DebuggerSettings.DynamicInstrumentationEnabled
-                || manager.ExceptionReplaySettings.Enabled)
+            var debuggerSettings = manager.DebuggerSettings;
+            bool diDisabled = !debuggerSettings.DynamicInstrumentationCanBeEnabled;
+            bool coDisabled = !debuggerSettings.CodeOriginForSpansCanBeEnabled;
+            bool erDisabled = !manager.ExceptionReplaySettings.CanBeEnabled || debuggerSettings.DynamicSettings.ExceptionReplayEnabled == false;
+
+            if (diDisabled && coDisabled && erDisabled)
```

Comment:
Just FYI, my concern is that these will almost never all be set (because customers would have to explicitly disable all of them) so this effectively means we will be initializing for all customers - we'll need to check the execution benchmarks to check that this doesn't have an adverse impact in general 🙂

Reply:
That true but we also check later that SymDb is `true` which will be true only if DI is not `false` so even if the customer set only DI to `false`, we will not upload symbols.